test/Mono.Linker.Tests/Tests/GetDisplayNameTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Mono.Cecil;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.TestCasesRunner;
+
+namespace Mono.Linker.Tests
+{
+	[TestFixture]
+	public class GetDisplayNameTests
+	{
+		[TestCaseSource (nameof (GetMemberAssertionsAsArray), new object[] { typeof (GetDisplayNameTests) })]
+		public void TestGetDisplayName (IMemberDefinition member, CustomAttribute customAttribute)
+		{
+			// The only intention with these tests is to check that the language elements that could
+			// show up in a warning are printed in a way that is friendly to the user.
+			if (customAttribute.AttributeType.Name != nameof (DisplayNameAttribute))
+				throw new NotImplementedException ();
+
+			var expectedDisplayName = (string) customAttribute.ConstructorArguments[0].Value;
+			switch (member.MetadataToken.TokenType) {
+			case TokenType.TypeRef:
+			case TokenType.TypeDef:
+				var x = (member as TypeReference).GetDisplayName ();
+				Assert.AreEqual (expectedDisplayName, (member as TypeReference).GetDisplayName ());
+				break;
+			case TokenType.MemberRef:
+			case TokenType.Method:
+				var y = (member as MethodReference).GetDisplayName ();
+				Assert.AreEqual (expectedDisplayName, (member as MethodReference).GetDisplayName ());
+				break;
+			default:
+				throw new NotImplementedException ();
+			}
+
+		}
+
+		public static IEnumerable<object[]> GetMemberAssertionsAsArray (Type type)
+		{
+			return MemberAssertionsCollector.GetMemberAssertions (type).Select (v => new object[] { v.member, v.ca });
+		}
+
+		[DisplayName ("GetDisplayNameTests.A")]
```

Comment:
Please add tests for
- generic types definition including nested combinations
- ref/out methods
- ref return types
- generic methods inside generic types

It'd be also good to have tests for TypeReferences